quantmsrescore/alphapeptdeep.py

```
@@ -0,0 +1,931 @@
+    ObservedSpectrum
+
+    Raises
+    ------
+    UnsupportedSpectrumFiletypeError
+        If the file extension is not supported.
+
+    """
+    try:
+        spectra = OpenMSHelper.get_mslevel_spectra(file_name=str(spec_file), ms_level=2)
+    except ValueError:
+        raise exceptions.UnsupportedSpectrumFiletypeError(Path(spec_file).suffixes)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)